packages/tiled/lib/src/common/property.dart

```diff
@@ -88,15 +88,43 @@ class Property<T> {
   }
 }
 
+/// A wrapper for a Tiled property set
+///
+/// Accessing an int value
+/// ```dart
+/// properties.get<int>('foo') == 3
+/// ```
+///
+/// Accessing an int property:
+/// ```dart
+/// properties.getProp<IntProperty>('foo') ==
+///     IntProperty(name: 'foo', value: 3);
+/// ```
 class CustomProperties extends Iterable<Property> {
   static const empty = CustomProperties({});
 
+  /// The properties, indexed by name
   final Map<String, Property> byName;
 
   const CustomProperties(this.byName);
 
+  /// Get a property value by its name.
+  ///
+  /// [T] must be the type of the value, which depends on the property type.
+  /// The following Tiled properties map to the follow Dart types:
+  ///  - int property -> int
+  ///  - float property -> double
+  ///  - boolean property -> bool
+  ///  - string property -> string
+  ///  - color property -> ui.Color
+  ///  - file property -> string (path)
+  ///  - object property -> int (ID)
+  T get<T>(String name) {
+    return getProp(name).value as T;
+  }
+
   /// Get a typed property by its name
-  T named<T extends Property<dynamic>>(String name) {
+  T getProp<T extends Property<dynamic>>(String name) {
```

Comment:
We have a ban on abbreviations 😉

Reply:
Which is why I was saying "as<T>"....

`CustomProperties` could also just implement:

```dart
Property operator[] (String name) => byName[name];
```

Then you get:
```dart
properties['fu']; // Returns IntProperty!
```
And if you extend `Property` to have `T as<T>()` it looks & feels like other code in Flame.

```dart
properties['fu'].as<int>(); // Returns integer!
```

Reply:
In that case, wouldn't it be smart to not have `CustomProperties` class at all. The internal `byName` map will also provide similar API.

----

> ```dart
> properties['fu']; // Returns IntProperty!
> ```

Technically, it will return `Property`. Users will have to manually cast it to `IntProperty`. In that regard, `properties.getProp<IntProperty>('foo')` feels much better.

--- 
We should also try to think about making it easier for end users to find these APIs. We can document the overloaded `[]` operator and `as<>()` as much as we want, but still there will be people who will not read the docs 😅. I feel that explicitly defined methods are easier to find because of code completion in IDEs.

Reply:
I think the overloaded operator looks better, I don't think users will have much more of a problem understanding that one than to see that there is the possibility to put generics on `getProp` (which should be named `getProperty` if we decide to go with that API anyways).

Is it it possible to put generics on the operator instead of using `as`?

Reply:
> In that case, wouldn't it be smart to not have `CustomProperties` class at all. The internal `byName` map will also provide similar API.
> 
> 
> 
> ----
> 
> 
> 
> > ```dart
> 
> > properties['fu']; // Returns IntProperty!
> 
> > ```
> 
> 
> 
> Technically, it will return `Property`. Users will have to manually cast it to `IntProperty`. In that regard, `properties.getProp<IntProperty>('foo')` feels much better.


I think the idea is that `as<int>` could be called on the base class so you don't have to cast the property first

> 
> 
> --- 
> 
> We should also try to think about making it easier for end users to find these APIs. We can document the overloaded `[]` operator and `as<>()` as much as we want, but still there will be people who will not read the docs 😅. I feel that explicitly defined methods are easier to find because of code completion in IDEs.

This was my primary reason for not just using Map style interface in the first place, but we could support both. However, `Property<T>.as<int>‘ returning the value itself feels like it's mixing levels of abstraction, it's operating on the value not in the property, is that weird?

Reply:
Operator[] is more natural to me. It can't have genetics to my knowledge.

Operator[] retiring the super instance `property` and the user casting to IntProperty (or runtime testing) is also pretty natural. This also grants access to the new custom classes in unparsed form maybe?


as<T> I feel belongs to property and not the map holding the properties.

Reply:
At that point, why not just `properties["foo"].value as int` why have a special method?